loopdb/batch.go

```diff
@@ -0,0 +1,51 @@
+package loopdb
+
+import (
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+type Batch struct {
+	// ID is the unique identifier of the batch.
+	ID lntypes.Hash
+
+	// BatchState is the current state of the batch.
+	BatchState string
+
+	// BatchTxid is the txid of the batch transaction.
+	BatchTxid chainhash.Hash
+
+	// BatchPkScript is the pkscript of the batch transaction.
+	BatchPkScript []byte
+
+	// LastRbfHeight is the height at which the last RBF attempt was made.
+	LastRbfHeight int32
+
+	// LastRbfSatPerKw is the sat per kw of the last RBF attempt.
+	LastRbfSatPerKw int32
+
+	// MaxTimeoutDistance is the maximum timeout distance of the batch.
+	MaxTimeoutDistance int32
```

Comment:
Doesn't `MaxTimeout` mostly refer to the "max timeout height" rather than the timeout difference of 2 sweeps?

Reply:
The comment wasn't helpful for me to understand that this mean pair-wise "timeout difference". Why would we measure that vs just max timeout of a batch?

Reply:
Because the upper limit here is not the only limit. You could have sweeps requesting to join a batch that have smaller timeout values

I will try to explain with an example:
- A sweep batch exists, which contains sweeps with timeouts `310`, `320`, `330`
- The `MaxTimeoutDistance` is set to `30`
- A new sweep requests to join a batch, which has a timeout of `299`
- The timeout distance of `299` and `330` is `31` > `MaxTimeoutDistance`

If we use `MaxTimeout` we would also have to use a `MinTimeout` and the values of both could dynamically change based on each sweep that enters a batch